### PR TITLE
Ports : Add epsilon 15.5.0

### PR DESCRIPTION
**Pull request by RedGl0w**
Reverted https://github.com/SerenityOS/serenity/commit/15d5c629158fd598d162f5acba26ffee7a923ff8 for libpng usage in epsilon
image

### CODE REVIEW

File: Userland/Libraries/LibC/assert.h [hunk 1]
```diff
@@ -14,11 +14,11 @@ __BEGIN_DECLS
...
-#    define assert(expr)                                                            \
-        (__builtin_expect(!(expr), 0)                                               \
-                ? __assertion_failed(#expr "\n" __FILE__ ":" __stringify(__LINE__)) \
-                : void(0))
-
+#    define assert(expr)                                                           \
+        do {                                                                       \
+            if (__builtin_expect(!(expr), 0))                                      \
+                __assertion_failed(#expr "\n" __FILE__ ":" __stringify(__LINE__)); \
+        } while (0)
```
Comment (BertalanD) [inline Userland/Libraries/LibC/assert.h:21]: I'm not sure about this change. It's kind of sad to lose the nice property of it being a single expression.

Can we do something like a normal C-style cast? i.e.
```c
(void)0
```

Comment (linusg) [reply Userland/Libraries/LibC/assert.h:21]: yep, revert is not gonna happen. "single expression" is more than just a nice property, it's effectively a requirement for portability. Fixed on master, please remove the commit.

**Comment by gunnarbeutner** [comment]
This now has a conflict because assert was fixed in the master branch.

File: Userland/Libraries/LibC/assert.h (hunk 1 above)
Comment (RedGl0w) [reply Userland/Libraries/LibC/assert.h:21]: > I'm not sure about this change. It's kind of sad to lose the nice property of it being a single expression.

This has been fixed in master, going to update the PR
> Even if not, please change the commit message to match our guidelines: the first line should not be more than 72 characters long and it must start with a subsystem name.

I noticed this issue, but I was not sure about how the new name should be chosen.

File: Ports/epsilon/package.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=epsilon
+version=15.5.0
+files="https://github.com/numworks/epsilon/archive/refs/tags/${version}.tar.gz ${port}-${version}.tar.gz 38c3b6baaf00863bbd179bce5e9cc42bbdbd0cd485b5bf3bbf4473383591bf83"
+auth_type=sha256
+buildopts="PLATFORM=simulator TARGET=serenity SERENITY_INSTALL_ROOT=${SERENITY_INSTALL_ROOT}"
```
Comment (linusg) [inline Ports/epsilon/package.sh:6]: This should be called `makeopts`, which then makes the custom `build()` redundant

File: Ports/epsilon/package.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+depends="SDL2 libpng libjpeg"
+workdir="${port}-${version}"
```
Comment (linusg) [inline Ports/epsilon/package.sh:8]: This is redundant

```suggestion
```

File: Ports/epsilon/package.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+build() {
+    run make $buildopts -j2
+}
```
Comment (linusg) [inline Ports/epsilon/package.sh:12]:
```suggestion
```

File: Ports/epsilon/patches/serenity.patch [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
++  png_write_end(png, NULL);
++
++  png_free_data(png, info, PNG_FREE_ALL, -1); // -1 = all items
++  png_destroy_write_struct(&png, nullptr);
++  fclose(file);
++}
++
++}
++}
++}
```
Comment (linusg) [inline Ports/epsilon/patches/serenity.patch:327]: Please split this into multiple patches (not necessarily one per file, but in a way that makes sense)

**Review by linusg — CHANGES_REQUESTED** [review]
Please remove the space between `Ports` and `:` (this needs to be fixed in the commit linter)

File: Ports/epsilon/package.sh [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+version=15.5.0
+files="https://github.com/numworks/epsilon/archive/refs/tags/${version}.tar.gz ${port}-${version}.tar.gz 38c3b6baaf00863bbd179bce5e9cc42bbdbd0cd485b5bf3bbf4473383591bf83"
+auth_type=sha256
+makeopts="PLATFORM=simulator TARGET=serenity SERENITY_INSTALL_ROOT=${SERENITY_INSTALL_ROOT}"
+depends="SDL2 libpng libjpeg"
+
+
+install() {
+    run mkdir -p ${SERENITY_INSTALL_ROOT}/opt/
+    run cp output/release/simulator/serenity/epsilon.elf ${SERENITY_INSTALL_ROOT}/opt/epsilon
```
Comment (GMTA) [inline Ports/epsilon/package.sh:12]: Why not install to `/usr/local/bin/`, like other ports?

File: Ports/epsilon/package.sh [hunk 7]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=epsilon
+version=15.5.0
+files="https://github.com/numworks/epsilon/archive/refs/tags/${version}.tar.gz ${port}-${version}.tar.gz 38c3b6baaf00863bbd179bce5e9cc42bbdbd0cd485b5bf3bbf4473383591bf83"
+auth_type=sha256
+makeopts="PLATFORM=simulator TARGET=serenity SERENITY_INSTALL_ROOT=${SERENITY_INSTALL_ROOT}"
+depends="SDL2 libpng libjpeg"
+
+
```
Comment (GMTA) [inline Ports/epsilon/package.sh:9]: You might want to add a launcher as well!

File: Ports/epsilon/package.sh (hunk 6 above)
Comment (RedGl0w) [reply Ports/epsilon/package.sh:12]: Not [all ports](https://github.com/SerenityOS/serenity/blob/5c89305fc6fce3aeb50725341b95caa8e8da7e5c/Ports/Another-World/package.sh#L20) are installed in `/usr/loca/bin` (and I didn't even see one of them yet).
But I'm going to change that yep

**Comment by gunnarbeutner** [comment]
Looks good to me. I can hold off on merging this if you want to add a launcher, otherwise we can always do that later on. :)

**Review by gunnarbeutner — APPROVED** [review]

**Pull request merged**